### PR TITLE
[pytest/hash] Rewrite hash test

### PR DESCRIPTION
**Pull request by William-zx**
### Description of PR

Summary:
Made the following change:

Rewrite hash test. Test hash keys defined in test_fib.py instead of hash_test.py.

Sub PR of [Enhance hash test #1494](https://github.com/Azure/sonic-mgmt/pull/1494)

### Type of change

- [] Bug fix
- [] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### CODE REVIEW

File: ansible/roles/test/files/ptftests/hash_test.py [hunk 1]
```diff
@@ -52,13 +52,9 @@ def setUp(self):
...
-        self.test_hash_srcip = self.test_params.get('hash_srcip', True)
-        self.test_hash_dstip = self.test_params.get('hash_dstip', True)
-        self.test_hash_srcport = self.test_params.get('hash_srcport', True)
-        self.test_hash_dstport = self.test_params.get('hash_dstport', True)
-        self.test_hash_inport = self.test_params.get('hash_inport', False)
-        self.test_ipv4 = self.test_params.get('ipv4', True)
-        self.test_ipv6 = self.test_params.get('ipv6', True)
+        self.src_ip_interval = lpm.LpmDict.IpInterval(ip_address(self.src_ip_range[0]), ip_address(self.src_ip_range[1]))
+        self.dst_ip_interval = lpm.LpmDict.IpInterval(ip_address(self.dst_ip_range[0]), ip_address(self.dst_ip_range[1]))
+        self.hash_keys = self.test_params.get('hash_keys', ['src-ip'])
```
Comment (lguohan) [inline ansible/roles/test/files/ptftests/hash_test.py:57]: in original test, by default, we test srcip, dstip, srcport, dstport, can you keep it the same?

File: ansible/roles/test/files/ptftests/hash_test.py [hunk 2]
```diff
@@ -98,63 +83,19 @@ def check_hash(self, src_ip_range, dst_ip_range, ipv4=True):
...
-        # step 5: check randomizing in port
-        # TODO
-
-    def check_ip_route(self, in_port, sport, dport, src_ip_addr, dst_ip_addr,
-                       dst_port_list, ipv4=True):
-        if ipv4:
-            (matched_index, received) = self.check_ipv4_route(in_port, sport, dport,
-                    src_ip_addr, dst_ip_addr, dst_port_list)
+        for _ in range(0, self.BALANCING_TEST_TIMES):
+            logging.info("src_port: {}".format(in_port))
```
Comment (lguohan) [inline ansible/roles/test/files/ptftests/hash_test.py:88]: src_port: -> ingress_port

File: ansible/roles/test/files/ptftests/hash_test.py [hunk 3]
```diff
@@ -98,63 +83,19 @@ def check_hash(self, src_ip_range, dst_ip_range, ipv4=True):
...
+            (matched_index, _) = self.check_ip_route(hash_key, in_port, dst_ip, exp_port_list)
+            hit_count_map[matched_index] = hit_count_map.get(matched_index, 0) + 1
+        logging.info("hit count map: {}".format(hit_count_map))
+        self.check_balancing(next_hop.get_next_hop(), hit_count_map)
+
+    def check_ip_route(self, hash_key, src_port, dst_ip, dst_port_list):
```
Comment (lguohan) [inline ansible/roles/test/files/ptftests/hash_test.py:94]: src_port -> in_port

File: ansible/roles/test/files/ptftests/hash_test.py [hunk 4]
```diff
@@ -98,63 +83,19 @@ def check_hash(self, src_ip_range, dst_ip_range, ipv4=True):
...
+        if ip_network(unicode(dst_ip)).version == 4:
+            (matched_index, received) = self.check_ipv4_route(hash_key, src_port, dst_port_list)
```
Comment (lguohan) [inline ansible/roles/test/files/ptftests/hash_test.py:96]: src_port -> in_port

File: tests/fib/test_fib.py [hunk 5]
```diff
@@ -93,70 +101,72 @@ def test_fib(testbed, duthost, ptfhost, ipv4, ipv6, mtu):
                 log_file=log_file,
                 socket_recv_size=16384)
 
-@pytest.mark.parametrize("ipv4, ipv6", [pytest.param(True, True)])
-def test_hash(testbed, duthost, ptfhost, ipv4, ipv6):
-
-    config_facts = duthost.config_facts(host=duthost.hostname, source="running")['ansible_facts']
 
+class Test_Hash():
```
Comment (lguohan) [inline tests/fib/test_fib.py:105]: if you use camel notation, do not mix with understore. should be TestHash

File: ansible/roles/test/files/ptftests/hash_test.py [hunk 6]
```diff
@@ -191,22 +135,25 @@ def check_ipv4_route(self, in_port, sport, dport,
...
 
-    def check_ipv6_route(self, in_port, sport, dport,
-                         ip_src, ip_dst, dst_port_list):
+    def check_ipv6_route(self, hash_key, src_port, dst_port_list):
         '''
         @summary: Check IPv6 route works.
-        @param source_port_index: index of port to use for sending packet to switch
-        @param dest_ip_addr: destination IP to build packet with.
+        @param hash_key: hash key to build packet with.
+        @param src_port: index of port to use for sending packet to switch
```
Comment (lguohan) [inline ansible/roles/test/files/ptftests/hash_test.py:148]: can you rename it to in_port, src_port can be easily confused with src-port which is l4 source port.

**Comment by lguohan** [comment]
i am ok with the rewrite, but have some minor comments, can you address them?

**Comment by lguohan** [comment]
I will review other PR once this one is done.

File: ansible/roles/test/files/ptftests/hash_test.py (hunk 1 above)
Comment (William-zx) [reply ansible/roles/test/files/ptftests/hash_test.py]: > in original test, by default, we test srcip, dstip, srcport, dstport, can you keep it the same?

done.

File: ansible/roles/test/files/ptftests/hash_test.py (hunk 2 above)
Comment (William-zx) [reply ansible/roles/test/files/ptftests/hash_test.py]: > src_port: -> ingress_port

done.

File: ansible/roles/test/files/ptftests/hash_test.py (hunk 3 above)
Comment (William-zx) [reply ansible/roles/test/files/ptftests/hash_test.py]: > src_port -> in_port

done.

File: ansible/roles/test/files/ptftests/hash_test.py (hunk 4 above)
Comment (William-zx) [reply ansible/roles/test/files/ptftests/hash_test.py]: > src_port -> in_port

done.

File: tests/fib/test_fib.py (hunk 5 above)
Comment (William-zx) [reply tests/fib/test_fib.py]: > if you use camel notation, do not mix with understore. should be TestHash

done.

File: ansible/roles/test/files/ptftests/hash_test.py (hunk 6 above)
Comment (William-zx) [reply ansible/roles/test/files/ptftests/hash_test.py]: > can you rename it to in_port, src_port can be easily confused with src-port which is l4 source port.

done.

**Comment by William-zx** [comment]
> i am ok with the rewrite, but have some minor comments, can you address them?

All modifications have been completed.

**Pull request merged**